### PR TITLE
Document `SYMFONY_PROJECT_DEFAULT_ROUTE` and `SYMFONY_APPLICATION_DEFAULT_ROUTE`

### PR DESCRIPTION
**Pull request by tucksaun**
To be released with v4.16.4

### CODE REVIEW

File: cookbooks/env.rst [hunk 1]
```diff
@@ -203,6 +203,17 @@ will evaluate to an empty string like any other unset variable:
     When several routes match a rule, the first one wins, the user order is
     kept. There's no preference regarding protocols.
 
+  .. tip::
+
+    For complex multi-applications projects where several applications are
+    publicly reachable, but one needs to determine the current application
+    endpoint (for webhooks for examples) and the project endpoint (to send
+    emails for instance), an additional
+    ``SYMFONY_APPLICATION_DEFAULT_ROUTE_`` environment variables set is
```
Comment (fabpot) [inline cookbooks/env.rst:212]: I would add * at the end `SYMFONY_APPLICATION_DEFAULT_ROUTE_*`

File: cookbooks/env.rst [hunk 2]
```diff
@@ -173,11 +173,11 @@ will evaluate to an empty string like any other unset variable:
...
-* ``SYMFONY_DEFAULT_ROUTE_URL``: The default endpoint serving your application.
-  It can be used to avoid hard-coding domains that can be used to reach
-  non-production environments. Parts of the URL (scheme, domain, port and path)
-  are also exposed each in their own variable whose name starts with
-  ``SYMFONY_DEFAULT_ROUTE_`` followed by the name of the part.
+* ``SYMFONY_PROJECT_DEFAULT_ROUTE_URL``: The default endpoint serving your
+  project. It can be used to avoid hard-coding domains that can be used to
+  reach non-production environments. Parts of the URL (scheme, domain, port and
+  path) are also exposed each in their own variable whose name starts with
+  ``SYMFONY_PROJECT_DEFAULT_ROUTE_`` followed by the name of the part.
```
Comment (fabpot) [inline cookbooks/env.rst:180]: Do we list the part names somewhere?

File: cookbooks/env.rst [hunk 3]
```diff
@@ -203,6 +203,17 @@ will evaluate to an empty string like any other unset variable:
     When several routes match a rule, the first one wins, the user order is
     kept. There's no preference regarding protocols.
 
+  .. tip::
+
+    For complex multi-applications projects where several applications are
```
Comment (fabpot) [inline cookbooks/env.rst:208]: I would remove complex as this is true for any multi-app project, right?

File: cookbooks/env.rst [hunk 4]
```diff
@@ -203,6 +203,17 @@ will evaluate to an empty string like any other unset variable:
...
+    publicly reachable, but one needs to determine the current application
+    endpoint (for webhooks for examples) and the project endpoint (to send
+    emails for instance), an additional
+    ``SYMFONY_APPLICATION_DEFAULT_ROUTE_`` environment variables set is
+    available. The same rules are applied to determine their value but only
+    routes matching the current application are evaluated.
+
```
Comment (fabpot) [inline cookbooks/env.rst:215]: double blank lines

File: cookbooks/env.rst (hunk 1 above)
Comment (tucksaun) [reply cookbooks/env.rst]: 👍

File: cookbooks/env.rst (hunk 3 above)
Comment (tucksaun) [reply cookbooks/env.rst:208]: Well, I would say 50/50 so you're right, let's remove "complex"

File: cookbooks/env.rst (hunk 2 above)
Comment (tucksaun) [reply cookbooks/env.rst:180]: implicitly:
> scheme, domain, port and path

Comment (fabpot) [reply cookbooks/env.rst:180]: ok, maybe we can change it to:

Parts of the URL are also exposed as their own variables whose name starts with ``SYMFONY_PROJECT_DEFAULT_ROUTE_`` followed by the name of the part (``SCHEME``, ``DOMAIN``, ``PORT``, and ``PATH``).

File: cookbooks/env.rst [hunk 5]
```diff
@@ -173,11 +173,12 @@ will evaluate to an empty string like any other unset variable:
...
 
-* ``SYMFONY_DEFAULT_ROUTE_URL``: The default endpoint serving your application.
-  It can be used to avoid hard-coding domains that can be used to reach
-  non-production environments. Parts of the URL (scheme, domain, port and path)
-  are also exposed each in their own variable whose name starts with
-  ``SYMFONY_DEFAULT_ROUTE_`` followed by the name of the part.
+* ``SYMFONY_PROJECT_DEFAULT_ROUTE_URL``: The default endpoint serving your
+  project. It can be used to avoid hard-coding domains that can be used to
+  reach non-production environments. Parts of the URL are also exposed as their
+  own variables whose name starts with ``SYMFONY_PROJECT_DEFAULT_ROUTE_``
```
Comment (OskarStark) [inline cookbooks/env.rst:179]:
```suggestion
  own variables whose name starts with ``SYMFONY_PROJECT_DEFAULT_ROUTE_*``
```

like below

Comment (tucksaun) [reply cookbooks/env.rst:179]: but then they don't start with this prefix as this is more a glob or a regex

**Pull request merged**